webpack/assets/javascripts/bundle_novnc.js

```diff
@@ -48,24 +49,32 @@ function connectFinished() {
   showStatus('normal', __('Connected'));
 }
 
+function on_error(e) {
+  if (e.code == 1006) {
+      showStatus('failed', __('The connection was closed by the browser. please verify that the certificate authority is valid'));
+  }
+}
 $(document).on('ContentLoad', () => {
   const vncScreen = $('#noVNC_screen');
 
   if (vncScreen.length) {
-    $('#sendCtrlAltDelButton').on('click', sendCtrlAltDel);
 
+    $('#sendCtrlAltDelButton').on('click', sendCtrlAltDel);
     const protocol = $('#vnc').data('encrypt') ? 'wss' : 'ws';
-    const host = window.location.hostname;
+    const host = $('#vnc').attr('data-host') || window.location.hostname;
     const port = $('#vnc').attr('data-port');
-    const url = `${protocol}://${host}:${port}`;
+    const path = $('#vnc').attr('data-path');
     const password = $('#vnc').attr('data-password');
-
-    rfb = new RFB(vncScreen.get(0), url, {
-      credentials: {
-        password,
-      },
-    });
-
+    const token_protocol =  $('#vnc').attr('data-token-protocol');
+    const plain_protocol =  $('#vnc').attr('data-plain-protocol');
+    const url = (path) ? `${protocol}://${host}:${port}${path}`: `${protocol}://${host}:${port}` ;
+    const options = {};
+    if (password) { options.credentials = {password} };
+    if (token_protocol || plain_protocol){
+        options.wsProtocols = [token_protocol,plain_protocol];
```

Comment:
Can't it lead to a situation where `token_protocol` exists but `plain_protocol` doesn't? In this case, you still using both of them.